examples/all-clusters-app/esp32/main/CaseCommands.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/CASESessionManager.h>
+#include <app/server/Server.h>
+#include <lib/core/CHIPEncoding.h>
+#include <lib/shell/Commands.h>
+#include <lib/shell/Engine.h>
+#include <lib/shell/commands/Help.h>
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace chip::DeviceLayer;
+
+namespace chip {
+namespace Shell {
+
+Shell::Engine sSubShell;
+static NodeId sNodeId;
+
+static void OnConnected(void * context, OperationalDeviceProxy * deviceProxy)
+{
+    streamer_printf(streamer_get(), "Establish CASESession Success!\r\n");
+}
+Callback::Callback<OnDeviceConnected> sOnConnectedCallback(OnConnected, nullptr);
+static void OnConnectionFailure(void * context, PeerId peerId, CHIP_ERROR error)
+{
+    streamer_printf(streamer_get(), "Establish CASESession Failure!\r\n");
+}
+Callback::Callback<OnDeviceConnectionFailure> sOnConnectionFailureCallback(OnConnectionFailure, nullptr);
+
+void ConnectToNode(intptr_t arg)
+{
+    FabricInfo * fabricInfo                 = reinterpret_cast<FabricInfo *>(arg);
+    Server * server                         = &(chip::Server::GetInstance());
+    CASESessionManager * caseSessionManager = server->GetCASESessionManager();
+    if (caseSessionManager == nullptr)
+    {
+        ChipLogError(SecureChannel, "Can't get the CASESessionManager");
+        return;
+    }
+    caseSessionManager->FindOrEstablishSession(fabricInfo->GetPeerIdForNode(sNodeId), &sOnConnectedCallback,
+                                               &sOnConnectionFailureCallback);
+}
+
+CHIP_ERROR ConnectToNodeHandler(int argc, char ** argv)
+{
+    const FabricIndex fabricIndex = static_cast<FabricIndex>(strtoul(argv[0], nullptr, 10));
+    FabricInfo * fabricInfo       = Server::GetInstance().GetFabricTable().FindFabricWithIndex(fabricIndex);
+
+    if (fabricInfo == nullptr)
+    {
+        streamer_printf(streamer_get(), "Can't get fabric information from FabricIndex\r\n");
+        return CHIP_ERROR_NOT_FOUND;
+    }
+    sNodeId = static_cast<NodeId>(strtoul(argv[1], nullptr, 10));
```

Comment:
This does not allow trying to connect to more than one node, right?  At least not until the previous one finishes....

Reply:
Yes, but this PR is raised only for RIO support verification. I think the situation of connecting to more than one node is rare. Do we really need to support establishing CASESession to two nodes or more at the same time?

Reply:
If the use cases are OK with just one node being supported and the behavior being weird if two such commands are run quickly that's fine.

It might also be better to track whether we are executing one command (e.g. via whether our sNodeId is a valid value) and fail out with a useful message if someone tries to do a second one before we are done, so people don't get the weird behavior....

Reply:
Add a check for it. Now the device will not try to establish session to another node util the last connection is done.